### PR TITLE
Add user authentication service

### PR DESCRIPTION
**Pull request by rez-gitlab**
## Description
This PR adds authentication.

## Changes
- Added new service class
- Updated routes
- Added tests

## Testing
- [ ] Unit tests pass
- [ ] Integration tests pass

## Screenshots
_Add screenshots here for UI changes_

### CODE REVIEW

**Comment by rez-gitlab** [comment]
LGTM! Nice implementation 👍

**Comment by rez-gitlab** [comment]
Could we add some tests for edge cases?

```ruby
it 'handles nil input' do
  expect(subject.call(nil)).to be_nil
end
```

**Comment by rez-gitlab** [comment]
I've addressed the feedback. Ready for re-review.

**Comment by rez-gitlab** [comment]
Screenshot 2025-11-25 at 13 04 35


Also this !!!

File: app/services/authentication_service.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class AuthenticationService
```
Comment (georgekoltsov) [inline app/services/authentication_service.rb:3]: imageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimage

**Comment by georgekoltsov** [comment]
imageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimageimage
